filestore-api/src/main/java/hu/blackbelt/osgi/filestore/api/FilenameUtils.java

```diff
@@ -0,0 +1,27 @@
+package hu.blackbelt.osgi.filestore.api;
+
+import lombok.NonNull;
+
+public class FilenameUtils {
+    @NonNull
+    public static String makeValidFilename(String fileName) {
+        String fn = fileName.trim();
+        while (fn.contains("  ")) {
```

Comment:
Why not regex then? I assume every repeated space needs to be changed to a single one?

Reply:
Because its cannot work for 3 or 4 spaces well